envoy/formatter/substitution_formatter.h

```diff
@@ -32,7 +32,8 @@ class Formatter {
                              const Http::ResponseHeaderMap& response_headers,
                              const Http::ResponseTrailerMap& response_trailers,
                              const StreamInfo::StreamInfo& stream_info,
-                             absl::string_view local_reply_body) const PURE;
+                             absl::string_view local_reply_body,
+                             absl::string_view access_log_type) const PURE;
```

Comment:
enum would be better than absl::string_view for this log type.

Reply:
Generally I agree, but I was trying to refrain adding reference to access_log.h here, as it seems that the formatter generally shouldn't have dependency on it, therefore the string_view

Reply:
HCM is name of HTTP L4 proxy, it's thing about the inner impl. I think it's would be better not to expose it to end users.

Reply:
Is this one meant for the other comment? I'm not ruling out renaming to HttpNewRequest, but then what would be the name for the router enum values? The router is also Http.

Reply:
>  as it seems that the formatter generally shouldn't have dependency on it

If we think the formatter generally shouldn't deps on it, then we shouldn't add this new parameter.

But now, to make the logger could support the %ACCESS_LOG_TYPE% command, we have to add this new parameter. I have created this dependency in logic. So, it doesn't make sense to use the string view here instead of the enumeration class.

Reply:
Got it. Yeah I guess that makes sense, I was fearing that adding direct reference to ``access_log.h`` would have a push back from reviewers, but now that you see that fit, I'll adjust the implementation. I guess I can add it with the default value 'NotSet', so it doesn't change the implementation in locations of the code that directly call the formatter, such as local reply and router's header formatter, etc..